test/rockpaperscissors.js

```diff
@@ -0,0 +1,717 @@
+        const expiryDate = timestamp + DAY_24_HOUR_IN_SECS;
+
+        await truffleAssert.eventEmitted(txObj, 'PlayerOneReveals', (ev) => {
+
+            return  ev.player === playerOne &&
+                    ev.gameKey === gameKey &&
+                    ev.move.toString(10) === "0" &&
+                    ev.expiryDate.toString(10) === expiryDate.toString(10)
+        },'PlayerOneReveals event is emitted');
+
+        const game = await rcp.games(gameKey);
+        assert.strictEqual(game.playerOneMove.toString(10), ROCK.toString(10));
+        assert.strictEqual(game.step.toString(10), PLAYER_ONE_REVEAL);
+        assert.strictEqual(game.expiryDate.toString(10), expiryDate.toString(10));
+    });
+
+
+    it("Player two reveals and player two wins", async () => {
+        await rcp.movePlayerOne(gameKey, toBN(10), {from: playerOne, value: toBN(10)});
+        await rcp.movePlayerTwo(gameKey, playerTwoMoveHash, {from: playerTwo, value: toBN(10)});
+        await rcp.revealPlayerOne(playerOneSecretBytes32, ROCK, {from: playerOne});
+        const txObj = await rcp.revealPlayerTwo(gameKey, playerTwoSecretBytes32, PAPER, {from: playerTwo});
+
+        await truffleAssert.eventEmitted(txObj, 'PlayerTwoReveals', (ev) => {
+
+            return  ev.player === playerTwo &&
+                    ev.gameKey === gameKey &&
+                    ev.move.toString(10) === "1"
+        },'PlayerTwoReveals event is emitted');
+
+        await truffleAssert.eventEmitted(txObj, 'GamePayment', (ev) => {
+
+            return  ev.player === playerTwo &&
+                ev.gameKey === gameKey &&
+                ev.amount.toString(10) === "19"
+        },'GamePayment event is emitted');
+
+       await truffleAssert.eventEmitted(txObj, 'GamePayment', (ev) => {
+
+            return  ev.player === playerOne &&
+                ev.gameKey === gameKey &&
+                ev.amount.toString(10) === "1"
+        },'GamePayment event is emitted');
+
+        const game = await rcp.games(gameKey);
+        assert.strictEqual(game.stake.toString(10), INIT);
+        assert.strictEqual(game.playerOne, playerOne);
+        assert.strictEqual(game.playerTwo, ZERO_ADDRESS);
+        assert.strictEqual(game.playerTwoMoveHash, ZERO_BYTES_32);
+        assert.strictEqual(game.playerOneMove.toString(10), "0");
+        assert.strictEqual(game.expiryDate.toString(10), "0");
+        assert.strictEqual(game.step.toString(10), INIT);
+
+        const playerOneOwed = await rcp.balances(playerOne);
+        const playerTwoOwed = await rcp.balances(playerTwo);
+
+        assert.strictEqual(playerOneOwed.toString(10), "1");
+        assert.strictEqual(playerTwoOwed.toString(10), "19");
+    });
+
+    it("Player two reveals and the game is a draw", async () => {
+        await rcp.movePlayerOne(gameKey, toBN(10), {from: playerOne, value: toBN(10)});
+        const playerTwoSameMoveHash = await rcp.createPlayerTwoMoveHash(gameKey, playerTwoSecretBytes32, ROCK, {from: playerTwo});
```

Comment:
You see how it would be easy to make a mistake here with `{from: playerTwo}`. That's why I am saying that a `view` function should not use `msg.sender`. It trips most people.

Reply:
OK. So I will pass in the player's address as a parameter.